src/context.rs

```diff
@@ -874,6 +878,41 @@ impl Context {
         }
     }
 
+    /// Evicts persistent objects or allows certain transient objects
+    /// to be made peristent.
+    pub fn evict_control(
+        &mut self,
+        auth: Provision,
+        object_handle: ObjectHandle,
+        persistent: Persistent,
+    ) -> Result<ObjectHandle> {
+        // A session for authorization is required.
+        if self.sessions.0.is_none() {
+            error!("Missing session handle for authorization of auth.");
+            return Err(Error::local_error(ErrorKind::InvalidParam));
```

Comment:
I'm wondering if this deserves its own error kind, invalid param seems a bit far fetched and I'm sure there are other API methods where you need at least one session.

Reply:
There are many of those, yeah. I think it might be an idea to make this a standard non-public `fn Context.require_a_session(&self) -> Result<()>` function.

Reply:
Yeah, I will look into a new error type and see how to make it easy to use in context.